odoo/service/server.py

```diff
@@ -1204,35 +1231,46 @@ def check_limits(self):
             _logger.info('WorkerCron (%s) max age (%ss) reached.', self.pid, config['limit_time_worker_cron'])
             self.alive = False
 
-    def _db_list(self):
-        if config['db_name']:
-            db_names = list(config['db_name'])
-        else:
-            db_names = odoo.service.db.list_dbs(True)
-        return db_names
-
     def process_work(self):
+        """Process a single database."""
         _logger.debug("WorkerCron (%s) polling for jobs", self.pid)
-        db_names = self._db_list()
-        if len(db_names):
-            self.db_index = (self.db_index + 1) % len(db_names)
-            db_name = db_names[self.db_index]
-            self.setproctitle(db_name)
-
-            from odoo.addons.base.models import ir_cron  # noqa: PLC0415
-            ir_cron.IrCron._process_jobs(db_name)
-
-            # dont keep cursors in multi database mode
-            if len(db_names) > 1:
-                odoo.sql_db.close_db(db_name)
-
-            self.request_count += 1
-            if self.request_count >= self.request_max and self.request_max < len(db_names):
-                _logger.error("There are more dabatases to process than allowed "
-                              "by the `limit_request` configuration variable: %s more.",
-                              len(db_names) - self.request_max)
+        # check notified databases
+        notified_databases = OrderedSet(notif.payload for notif in self.dbcursor.notifies if notif.channel == 'cron_trigger')
+        self.dbcursor.notifies.clear()  # free resources
+
+        # list databases
+        if not self.db_queue:
+            database_list = cron_database_list()
+            self.db_queue.update(reversed(database_list))
+            self.db_count = len(self.db_queue)
         else:
-            self.db_index = 0
+            # move notified databases as to be processed soon
+            # ignore unqueued databases
+            for db_name in notified_databases:
```

Comment:
`self.dbcursor.notifies` can be treated as a FIFO
In `notified_databases`, elements with small index are for earlier `cron_trigger`
`self.db_queue` is a LIFO, should we reverse `notified_databases` here to allow earlier `cron_trigger` to be handled earlier?